### ISSUE TITLE
[tune] adjusting custom trial names

### ISSUE THREAD

**Issue by ninafiona** [ray-project/ray#6953]
I want to change my trial name, as suggested here:
https://ray.readthedocs.io/en/latest/tune-usage.html#custom-trial-names

Example (from https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/logging_example.py)

```
def trial_name_string(trial):
    return "{}_{}_123".format(trial.trainable_name, trial.trial_id)
```

Are there more options for changing the name? I tried custom return values (like str(trial).split('-')[0] or just "Pendulumv0") and got errors or the trial name remained default. I also want to create an 'info.txt' file in that directory, can I access the trial name outside of the tune.run() function?

similar issue #3034

**Comment by richardliaw** [comment]
Can you provide an issue for reproduction for what you tried and what errors you got? 

Also you should be able to access trial names as:

```
analysis = tune.run(...)
names = [str(trial) for trial in analysis.trials]

**Comment by ninafiona** [comment]
Thank you @richardliaw, I was able to access the trial names and create files into the directories (I think it would be better to create the files before running the tune.run(), in case there is an error coming up etc, but it's better than nothing.)

I wasn't able to reproduce the error for the trial name, the code below is running without any errors now. The resulting filename is "SAC_Pendulum-v0_MLP_2020-02-03_14-00-56g21_qlu9", so the date and id are always added automatically. I tried to call my info_to_file() function inside the trial_str_creator to save the info file before running the algorithm, but the trial_id isn't the same in that case.

`def trial_str_creator(trial):
    trialname = algorithm + "_" + environment + "_" + model 
    info_to_file(trialname + "_" + time.strftime("%Y-%m-%d_%H-%M-%S") + trial.trial_id)
    return trialname

ray.init()
analysis = tune.run(
    algorithm,
    local_dir=output_dir,
    stop={"episode_reward_mean": -200},
    config=config,
    trial_name_creator=trial_str_creator
)`
